packages/build-utils/src/get-project-paths.ts

```diff
@@ -0,0 +1,67 @@
+import { detectFramework } from './detect-framework';
+import { DetectorFilesystem } from './detectors/filesystem';
+import frameworks from '@vercel/frameworks';
+
+const MAX_DEPTH_TRAVERSE = 3;
```

Comment:
How/why was 3 chosen?

Reply:
To some degree the decision that we make here actually informs the decisions that people make about project layout. I don't suspect people are particularly dogmatic about these things.